activerecord/test/cases/view_test.rb

```diff
@@ -193,6 +193,12 @@ def test_insert_record
         assert_equal "Rails in Action", new_book.name
       end
 
+      def test_insert_record_populates_primary_key
+        book = PrintedBook.create! name: "Rails in Action", status: 0, format: "paperback"
+        assert_not_nil book.id
+        assert book.id > 0
+      end if current_adapter?(:PostgreSQLAdapter, :SQLite3Adapter) && supports_insert_returning?
```

Comment:
Just to clarify, this condition is not a limitation of the database/adapter, at least not for mysql, but to make Views work for mysql we need a separate fix which may end up being tricky and I don't want to block this fix which still addresses two of the reported problems